packages/lisk-p2p/src/peer_pool.ts

```diff
@@ -253,33 +256,74 @@ export class PeerPool extends EventEmitter {
 		});
 	}
 
+	public async fetchStatusAndCreatePeer(
+		seedPeers: ReadonlyArray<P2PPeerInfo>,
+		nodeInfo: P2PNodeInfo,
+		peerConfig: PeerConfig,
+	): Promise<ReadonlyArray<P2PDiscoveredPeerInfo>> {
+		const seedPeerUpdatedInfosAndSocket = await Promise.all(
+			seedPeers.map(async seedPeer =>
+				connectAndFetchPeerInfo(seedPeer, nodeInfo, peerConfig),
+			),
+		);
+		// We can log the errors returned but it will not fail if one of the calls failed
+		const listOfPeerInfos = seedPeerUpdatedInfosAndSocket.reduce<
+			ReadonlyArray<P2PDiscoveredPeerInfo>
+		>((peerInfoList, seedPeerResponse) => {
+			if (!(seedPeerResponse instanceof Error)) {
+				const peerId = constructPeerIdFromPeerInfo(seedPeerResponse.peerInfo);
+
+				this.addOutboundPeer(
+					peerId,
+					seedPeerResponse.peerInfo,
+					seedPeerResponse.socket,
+				);
+
+				return [...peerInfoList, seedPeerResponse.peerInfo];
+			}
+
+			return peerInfoList;
+		}, []);
+
+		return listOfPeerInfos;
+	}
+
 	public async runDiscovery(
 		knownPeers: ReadonlyArray<P2PDiscoveredPeerInfo>,
 		blacklist: ReadonlyArray<P2PPeerInfo>,
 	): Promise<ReadonlyArray<P2PDiscoveredPeerInfo>> {
-		const peersObjectList = knownPeers.map(
-			(peerInfo: P2PDiscoveredPeerInfo) => {
-				const peerId = constructPeerIdFromPeerInfo(peerInfo);
-				if (this.hasPeer(peerId)) {
-					const existingPeer = this.getPeer(peerId) as Peer;
-					existingPeer.updatePeerInfo(peerInfo);
-
-					return existingPeer;
-				}
-
-				return this.addPeer(peerInfo);
-			},
-		);
+		const peersForDiscovery = knownPeers.map(peerInfo => {
+			const peerId = constructPeerIdFromPeerInfo(peerInfo);
+			if (this.hasPeer(peerId)) {
+				const existingPeer = this.getPeer(peerId) as Peer;
+
+				existingPeer.updatePeerInfo(peerInfo);
+
+				return existingPeer;
+			}
+
+			return this.addPeer(peerInfo);
+		});
 
 		const peerSampleToProbe = selectRandomPeerSample(
-			peersObjectList,
+			peersForDiscovery,
 			MAX_PEER_DISCOVERY_PROBE_SAMPLE_SIZE,
 		);
 
 		const discoveredPeers = await discoverPeers(peerSampleToProbe, {
 			blacklist: blacklist.map(peer => peer.ipAddress),
 		});
 
+		// Check for received discovery info and then find it in peer pool and then update it
+		discoveredPeers.forEach((peerInfo: P2PDiscoveredPeerInfo) => {
+			const peerId = constructPeerIdFromPeerInfo(peerInfo);
+			if (this.hasPeer(peerId)) {
```

Comment:
instead of calling `hasPeer` and casting `getPeer` result, maybe just call `getPeer` and check undefined?